template/map.jinja

```diff
@@ -9,16 +9,44 @@
 {%- import_yaml tplroot ~ "/osmap.yaml" as osmap %}
 {%- import_yaml tplroot ~ "/osfingermap.yaml" as osfingermap %}
 
-{%- set defaults = salt['grains.filter_by'](default_settings,
-    default='template',
-    merge=salt['grains.filter_by'](osfamilymap, grain='os_family',
-      merge=salt['grains.filter_by'](osmap, grain='os',
-        merge=salt['grains.filter_by'](osfingermap, grain='osfinger',
-          merge=salt['config.get']('template:lookup', default={})
+{%- if salt['config.get']('__cli') == 'salt-call' %}
```

Comment:
We may run into problems because `salt-call` and `salt-ssh` are two different ways of using Salt.
But maybe `defaults.merge` is broken for `salt-call` too, so this approach would be fine.

```yaml
# salt-call grains.get saltpath
local:
    /usr/local/lib/python3.6/site-packages/salt

# salt-call config.get __cli   
local:
    salt-call
```

```yaml
#% salt-ssh host grains.get saltpath
host:
    /var/tmp/.root_99b0b6_salt/pyall/salt

#% salt-ssh host config.get __cli   
host:
    salt-call
```

Reply:
This is a good point, which has helped establish that `config.get __cli` is not good enough by itself to identify `salt-ssh`.  Just tried `salt-call` on a standard minion and it worked fine with `defaults.merge`, so it shouldn't lose out by being batched alongside `salt-ssh` (if we do have to use `defaults.merge` as the solution).

I've got some `config.items` diffs between all of the types, so I'm sure we can find a reliable solution (if required).

Reply:
(my 2 cents) `salt-call` can also be used to run salt commands from a minion and not a master, but with the same results I think.